src/main/java/hudson/plugins/locale/PluginImpl.java

```diff
@@ -134,6 +137,13 @@ public GlobalConfigurationCategory getCategory() {
         return GlobalConfigurationCategory.get(AppearanceCategory.class);
     }
 
+    public List<Locale> getLocales() {
```

Comment:
I would use here the approach as described here https://weekly.ci.jenkins.io/design-library/Select/ with the `f:select` and implement  `doFillSystemLocale`